### PR TITLE
fix(porting): porting Activity fixes and SMS fixes from last release

### PR DESCRIPTION
**Pull request by rowasc**
This pull request makes the following changes:
-

Test checklist:
- [ ]

- [ ] I certify that I ran my checklist

Fixes ushahidi/platform# .

Ping @ushahidi/platform

### CODE REVIEW

**Comment by rowasc** [comment]
Still Missing in master->dev port: https://github.com/ushahidi/platform/pull/3236/files (session expiration)

File: app/Http/Controllers/API/CSV/CSVImportController.php [hunk 1]
```diff
@@ -29,31 +29,18 @@ public function store(Request $request, $id = null)
...
-
-        // Set map and fixed values for transformer
-        $transformer->setMap($csv->maps_to);
-        $transformer->setFixedValues($csv->fixed);
-
         $this->usecase = $this->usecaseFactory
             ->get($this->getResource(), 'import')
-            ->setPayload($records)
-            ->setTransformer($transformer);
+            //->setPayload($records)
```
Comment (willdoran) [inline app/Http/Controllers/API/CSV/CSVImportController.php]: delete the commented lines if we don't need them

File: app/Http/Controllers/API/CSV/CSVImportController.php [hunk 2]
```diff
@@ -29,31 +29,18 @@ public function store(Request $request, $id = null)
...
+            ->setCSV($csv);
+            //->setTransformer($transformer);
```
Comment (willdoran) [inline app/Http/Controllers/API/CSV/CSVImportController.php]: delete this too

File: app/Http/Controllers/API/CSV/CSVImportController.php [hunk 3]
```diff
@@ -29,31 +29,18 @@ public function store(Request $request, $id = null)
          * (primarily added because of OS9 line endings which do not work by default )
          */
         ini_set('auto_detect_line_endings', 1);
+        ini_set('memory_limit', '-1');
+        set_time_limit(720);
+
```
Comment (willdoran) [inline app/Http/Controllers/API/CSV/CSVImportController.php]: Do we need these two if we are doing it in the job?

File: src/Core/Entity/FormStats.php [hunk 4]
```diff
@@ -21,6 +21,7 @@ class FormStats extends StaticEntity
     protected $total_messages_sent;
     protected $total_messages_pending;
     protected $total_by_data_source;
+
```
Comment (willdoran) [inline src/Core/Entity/FormStats.php]: I don't think this file needs to change

File: src/Core/Usecase/ImportUsecase.php [hunk 5]
```diff
@@ -12,12 +12,17 @@
...
 use Traversable;
+use Ushahidi\App\Jobs\ImportJob;
 use Ushahidi\Core\Entity;
 use Ushahidi\Core\Usecase;
 use Ushahidi\Core\Tool\AuthorizerTrait;
 use Ushahidi\Core\Tool\FormatterTrait;
 use Ushahidi\Core\Tool\ValidatorTrait;
 use Ushahidi\Core\Tool\Transformer;
+use Ushahidi\Core\Entity\CSV;
+use League\Event\ListenerInterface;
```
Comment (willdoran) [inline src/Core/Usecase/ImportUsecase.php]: I think this isn't needed

File: src/Core/Usecase/ImportUsecase.php [hunk 6]
```diff
@@ -12,12 +12,17 @@
...
+use Ushahidi\Core\Traits\Event;
```
Comment (willdoran) [inline src/Core/Usecase/ImportUsecase.php]: I think this isn't needed

File: src/Core/Usecase/ImportUsecase.php [hunk 7]
```diff
@@ -27,11 +32,19 @@ class ImportUsecase implements Usecase
     use AuthorizerTrait,
         FormatterTrait,
         ValidatorTrait;
-
+    // Use Event trait to trigger events
+    use Event;
```
Comment (willdoran) [inline src/Core/Usecase/ImportUsecase.php]: I think this isn't needed

File: src/Core/Usecase/ImportUsecase.php [hunk 8]
```diff
@@ -96,53 +109,37 @@ public function isSearch()
...
         // ... and return the formatted entity
         return [
-            'created_ids' => $created_entities,
-            'processed' => $processed,
-            'errors' => $errors
+            'status' => $new_status
         ];
     }
 
+    public function verify($entity)
```
Comment (willdoran) [inline src/Core/Usecase/ImportUsecase.php]: I think this isn't needed

File: app/Http/Controllers/API/CSV/CSVImportController.php (hunk 3 above)
Comment (rowasc) [reply app/Http/Controllers/API/CSV/CSVImportController.php]: oh yea we dont. nice catch

File: src/Core/Entity/FormStats.php (hunk 4 above)
Comment (rowasc) [reply src/Core/Entity/FormStats.php]: oh. phpcs was complaining about spacing that's why it changed

**Pull request merged**